app/models/web_search.rb

```diff
@@ -79,15 +79,18 @@ def run_odie_search_and_handle_response(odie_search, available_search_engine_pag
     if odie_response&.total&.positive?
       adjusted_total = (available_search_engine_pages * @per_page) + odie_response.total
       if @total <= @per_page * (@page - 1) && available_search_engine_pages < @page
-        paginate_odie_response(odie_search, odie_response)
+        paginate_odie_response(odie_search, odie_response, adjusted_total)
       end
       @total = adjusted_total
     end
     odie_response
   end
 
-  def paginate_odie_response(odie_search, odie_response)
+  def paginate_odie_response(odie_search, odie_response, adjusted_total)
+    temp_total = @total
```

Comment:
The primary code change in this PR is that lns 90-96 have been reverted to what they were prior to SRCH-4075 (though this might not be obvious at first since SRCH-4075 broke this code out into its own `paginate_odie_response` method to reduce the complexity of the original `run_odie_search_and_handle_response` method).  See: https://github.com/GSA/search-gov/blob/6f7530171bfd96c65cd682a9dad0fbcfc1bb1842/app/models/web_search.rb#L78-L84